InformationScripting/src/query_prompt/QueryBuilder.cpp

```diff
@@ -89,7 +95,14 @@ Query* QueryBuilder::visitOperator(QueryBuilder* self, OperatorQueryNode* op)
 	else if (leftComposite)
 	{
 		// union
-		connectQueriesWith(composite, leftComposite, new UnionOperator(), right);
+		Query* unionQuery = nullptr;
+		if (op->op() == OperatorQueryNode::OperatorTypes::Pipe)
+			unionQuery = composite->addQuery(std::unique_ptr<Query>(new UnionOperator()));
```

Comment:
Yes the pipe operator also performs a split, but this UnionOperator is only for unions. A split and a normal pipe are only implicit, since they just describe a wiring in the composite operator, but for the union we need to take some special action.

Even if the user only thinks about a Pipe I think at this level it makes to talk about a union, since that is the only thing that this operator does.

But if you really want to name it differently we can do so, but I guess in some way it still will need union in the name, otherwise someone in the future might think: "Oh cool a pipe operator, let's use it for a split" and it won't work as expected.